src/transactions/cdp.js

```diff
@@ -385,14 +364,161 @@ export async function openCDP(openingALGOs, openingGARD, commit, toWallet) {
     stxns.push(stxn8.blob)
   }
   
+  return stxns
+}
+
+async function openASACDP(openingMicroAssetAmount, microOpeningGard, asaID, info, accountID, cdp) {
+  const paramsPromise = getParams(2000);
+  
+  // Part 1: Opting in, creating needed info, etc.
+
+  /*
+  if (
+    307000 +
+      openingMicroAssetAmount +
+      100000 * (info["assets"].length + 4) >
+    info["amount"]
+  ) {
+    return {
+      alert: true,
+      text:
+        "Depositing this much collateral will put you below your minimum balance.\n" +
+        "Your Maximum deposit is: " +
+        (info["amount"] -
+          307000 -
+          100000 * (info["assets"].length + 4)) /
+          1000000 +
+        " Algos",
+    };
+  }
+  */ // TODO: Fix this
+
+  let params = await paramsPromise;
+  let txns = makeOptInTxns(info, params);
+  let optins = txns.length;
+  params.fee = 7000;
+  // next two txns
+  txns = txns.concat(_openCDPtxns1(850000, cdp, info, params)) // XXX: This amount may not be optimized.
```

Comment:
If addresses are re-used. We shouldn't be doing this 850,000. Let's add a check in the function for current cdp balance?